### DISCUSSION TITLE
What's the best way to setup prisma middleware with Nextjs?

### DISCUSSION THREAD

**Discussion by mmintel** [prisma/prisma#8356] [Q&A] [answered]
Hey there,

what's the best way to setup prisma middleware (`$use`) with Nextjs? Is it possible to do so without using a custom server? Asking because Nextjs splits the code into standalone functions and middleware should probably just be setup once for the whole server. Do I have to call them on every endpoint or it probably doesn't make sense at all to use them with Next?

Cheers

**Comment by ryands17** [comment, marked as answer]
@mmintel 👋 

Have a separate file for `PrismaClient` and its corresponding middlewares and import that in your functions. This way, you'll have the middlewares bundled along with each function.

**Reply by mmintel** [reply]
That's what I ended up with but great to know it's the recommended way, thanks!

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
